src/biokbase/narrative/jobs/appmanager.py

```diff
@@ -594,6 +612,34 @@ def _validate_bulk_app_info(self, app_info: dict):
                 f"an app version must be a string, not {app_info['version']}"
             )
 
+    def _reconstitute_shared_params(self, app_info_el: dict) -> None:
+        """
+        Mutate each params dict to include any shared_params
+        app_info_el is structured like:
+        {
+            "app_id": "Some_module/reads_to_contigset",
+            "tag": "release",
+            "version": "1.0.0",
+            "shared_params": {
+                "filter_len": 500
+            },
+            "params": [
+                {
+                    "read_library_name" : "My_PE_Library",
+                    "output_contigset_name" : "My_Contig_Assembly"
+                }, {
+                    "read_library_name": "Another_reads_library",
+                    "output_contigset_name": "Another_contig_assembly"
+                }
+            ]
+        }
+        """
+        if "shared_params" in app_info_el:
+            shared_params = app_info_el.pop("shared_params")
+            for param_set in app_info_el["params"]:
+                for k, v in shared_params.items():
+                    param_set.setdefault(k, v)
+
```

Comment:
I think it was mentioned that `shared_params` would always be around and would not overlap with the `param_set`s, but this is backwards compatible and future-proof if there ever is any overlap ...